tests/test_web_urldispatcher.py

```diff
@@ -407,3 +408,19 @@ async def test_static_absolute_url(aiohttp_client, tmp_path) -> None:
     client = await aiohttp_client(app)
     resp = await client.get('/static/' + str(file_path.resolve()))
     assert resp.status == 403
+
+def test_register_resource(aiohttp_client) -> None:
+    router = UrlDispatcher()
+    path = "/result/"
+    url = URL.build(path=path)
+
+    correct_name = "check-for-result"
+    resource = PlainResource(url.raw_path, name=correct_name)
+    router.register_resource(resource)
+
+    assert resource == router._named_resources[correct_name]
+
+    wrong_name = "for"
+    resource = PlainResource(url.raw_path, name=wrong_name)
+    with pytest.raises(ValueError):
```

Comment:
You have to use `match=` arg as show @ https://github.com/aio-libs/aiohttp/pull/4692#discussion_r411475905